transport/http2_client.go

```diff
@@ -496,10 +496,17 @@ func (t *http2Client) Close() (err error) {
 
 func (t *http2Client) GracefulClose() error {
 	t.mu.Lock()
-	if t.state == closing || t.state == unreachable {
```

Comment:
not really. ur suggestion is incorrect because unlock is required for t.Close().